File: src/main/scala/jk_5/nailed/worldedit/WorldEditEntityType.scala
package jk_5.nailed.worldedit

import com.google.common.base.Preconditions.checkNotNull
import com.sk89q.worldedit.entity.metadata.EntityType
import net.minecraft.entity.{Entity, EntityLiving, IMerchant, INpc, IProjectile}
import net.minecraft.entity.item.{EntityBoat, EntityEnderEye, EntityFallingBlock, EntityItem, EntityItemFrame, EntityMinecart, EntityPainting, EntityTNTPrimed, EntityXPOrb}
import net.minecraft.entity.monster.EntityGolem
import net.minecraft.entity.passive.{EntityAmbientCreature, EntityTameable, IAnimals}
import net.minecraft.entity.player.EntityPlayer

/**
 * No description given
 *
 * @author jk-5
 */
class WorldEditEntityType(val entity: Entity) extends EntityType {
  checkNotNull(entity)

  override def isPlayerDerived = entity.isInstanceOf[EntityPlayer]
  override def isProjectile = entity.isInstanceOf[EntityEnderEye] || entity.isInstanceOf[IProjectile]
  override def isItem = entity.isInstanceOf[EntityItem]
  override def isFallingBlock = entity.isInstanceOf[EntityFallingBlock]
  override def isPainting = entity.isInstanceOf[EntityPainting]
  override def isItemFrame = entity.isInstanceOf[EntityItemFrame]
  override def isBoat = entity.isInstanceOf[EntityBoat]
  override def isMinecart = entity.isInstanceOf[EntityMinecart]
  override def isTNT = entity.isInstanceOf[EntityTNTPrimed]
  override def isExperienceOrb = entity.isInstanceOf[EntityXPOrb]
  override def isLiving = entity.isInstanceOf[EntityLiving]
  override def isAnimal = entity.isInstanceOf[IAnimals]
  override def isAmbient = entity.isInstanceOf[EntityAmbientCreature]
  override def isNPC = entity.isInstanceOf[INpc] || entity.isInstanceOf[IMerchant]
  override def isGolem = entity.isInstanceOf[EntityGolem]
  override def isTamed = entity.isInstanceOf[EntityTameable] && entity.asInstanceOf[EntityTameable].isTamed
  override def isTagged = entity.isInstanceOf[EntityLiving] && entity.asInstanceOf[EntityLiving].hasCustomNameTag
}

File: src/main/java/jk_5/nailed/plugins/worldedit/WorldEditEntityType.java
package jk_5.nailed.plugins.worldedit;

import com.sk89q.worldedit.entity.metadata.EntityType;
import net.minecraft.entity.*;
import net.minecraft.entity.item.*;
import net.minecraft.entity.monster.EntityGolem;
import net.minecraft.entity.passive.EntityAmbientCreature;
import net.minecraft.entity.passive.EntityTameable;
import net.minecraft.entity.passive.IAnimals;
import net.minecraft.entity.player.EntityPlayer;

import static com.google.common.base.Preconditions.checkNotNull;

public class WorldEditEntityType implements EntityType {

    private final Entity entity;

    public WorldEditEntityType(Entity entity) {
        checkNotNull(entity);
        this.entity = entity;
    }

    @Override
    public boolean isPlayerDerived() {
        return entity instanceof EntityPlayer;
    }

    @Override
    public boolean isProjectile() {
        return entity instanceof EntityEnderEye || entity instanceof IProjectile;
    }

    @Override
    public boolean isItem() {
        return entity instanceof EntityItem;
    }

    @Override
    public boolean isFallingBlock() {
        return entity instanceof EntityFallingBlock;
    }

    @Override
    public boolean isPainting() {
        return entity instanceof EntityPainting;
    }

    @Override
    public boolean isItemFrame() {
        return entity instanceof EntityItemFrame;
    }

    @Override
    public boolean isBoat() {
        return entity instanceof EntityBoat;
    }

    @Override
    public boolean isMinecart() {
        return entity instanceof EntityMinecart;
    }

    @Override
    public boolean isTNT() {
        return entity instanceof EntityTNTPrimed;
    }

    @Override
    public boolean isExperienceOrb() {
        return entity instanceof EntityXPOrb;
    }

    @Override
    public boolean isLiving() {
        return entity instanceof EntityLiving;
    }

    @Override
    public boolean isAnimal() {
        return entity instanceof IAnimals;
    }

    @Override
    public boolean isAmbient() {
        return entity instanceof EntityAmbientCreature;
    }

    @Override
    public boolean isNPC() {
        return entity instanceof INpc || entity instanceof IMerchant;
    }

    @Override
    public boolean isGolem() {
        return entity instanceof EntityGolem;
    }

    @Override
    public boolean isTamed() {
        return entity instanceof EntityTameable && ((EntityTameable) entity).isTamed();
    }

    @Override
    public boolean isTagged() {
        return entity instanceof EntityLiving && ((EntityLiving) entity).hasCustomName();
    }

    //@Override //TODO
    public boolean isArmorStand() {
        return entity instanceof EntityArmorStand;
    }
}